lib/chef/provider/aws_rds_instance.rb

```diff
@@ -0,0 +1,49 @@
+require 'chef/provisioning/aws_driver/aws_provider'
+
+class Chef::Provider::AwsRdsInstance < Chef::Provisioning::AWSDriver::AWSProvider
+  REQUIRED_OPTIONS = %w(db_instance_identifier allocated_storage engine
+                        db_instance_class master_username master_user_password)
+
+  OTHER_OPTIONS = %w(engine_version multi_az iops publicly_accessible db_name db_port)
+
+  def update_aws_object(instance)
+    Chef::Log.warn("aws_rds_instance does not support modifying a started instance")
+  end
+
+  def create_aws_object
+    converge_by "Create new RDS instance #{new_resource.db_instance_identifier} in #{region}" do
+      new_resource.driver.rds.client.create_db_instance(create_options)
+    end
+  end
+
+  def destroy_aws_object(instance)
+    converge_by "Deleting RDS instance #{new_resource.db_instance_identifier} in #{region}" do
+      instance.delete(skip_final_snapshot: true)
+    end
+  end
+
+  def create_options
+    opts = {}
+    opts.merge(new_resource.additional_options)
+    REQUIRED_OPTIONS.each do |opt|
+      opts[opt] = new_resource.send(opt.to_sym)
+    end
+    OTHER_OPTIONS.each do |opt|
+      opts[opt] = new_resource.send(opt.to_sym) if ! new_resource.send(opt.to_sym).nil?
+    end
+    opts[:tags] = hash_to_tag_array(new_resource.aws_tags) if new_resource.aws_tags
```

Comment:
You shouldn't _need_ to do this because the provider [takes care](https://github.com/chef/chef-provisioning-aws/blob/master/lib/chef/provisioning/aws_driver/aws_provider.rb#L124) of it for you.

Pros of this - having this here reduces the number of API calls we make to AWS
Cons of this - It adds more duplicated tagging code, of which there is already a lot ([1](https://github.com/chef/chef-provisioning-aws/blob/master/lib/chef/provisioning/aws_driver/aws_provider.rb#L237), [2](https://github.com/chef/chef-provisioning-aws/blob/master/lib/chef/provisioning/aws_driver/driver.rb#L94-L101), [3](https://github.com/chef/chef-provisioning-aws/blob/master/lib/chef/provisioning/aws_driver/driver.rb#L284-L311), [4](https://github.com/chef/chef-provisioning-aws/blob/master/lib/chef/provisioning/aws_driver/driver.rb#L1080))

Because my primary concern at this point is code complexity I would rather remove this logic and let the provider handle tagging.  Unless that wasn't working somehow?